### PR TITLE
Add a blub to README about running YACE as a library

### PR DESCRIPTION
**Pull request by kgeckhart**
Looking for some feedback on this before I submit it upstream with the metric registration changes. I'm not always the best at writing coherent docs.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,6 +33,7 @@ We will contact you as soon as possible.
 * Allows exports metrics with CloudWatch timestamps (disabled by default)
 * Static metrics support for all cloudwatch metrics without auto discovery
 * Pull data from multiple AWS accounts using cross-account roles
+* Can be used as a library in an existing application
```
Comment (matthewnolf) [inline README.md]:
```suggestion
* Can be used as a library in an external application
```
minor thought: maybe external

File: README.md [hunk 2]
```diff
@@ -582,6 +583,26 @@ This protects from the abuse of API requests that can cause extra billing in AWS
 The flag 'scraping-interval' defines the seconds between scrapes.
 The default value is 300.
 
+### Embedding YACE as a library in an existing application
+It is possible to embed YACE in to an existing library to scrape on demand. This might be useful to you would like to scrape on demand or run in a stateless manner.
+
+The entrypoint to use YACE as a library is [update.go](./pkg/update.go) which requires,
```
Comment (matthewnolf) [inline README.md]: Maybe worth mentioning the entrypoint func, rather than file name (or both?)

Comment (matthewnolf) [reply README.md]: A link to GoDoc here may also be nice

File: README.md [hunk 3]
```diff
@@ -582,6 +583,26 @@ This protects from the abuse of API requests that can cause extra billing in AWS
...
+- `cloudwatchSemaphore`/`tagSemaphore`: adjusts the concurrency concurrency of requests as defined by [Requests concurrency](#requests-concurrency). Pass in a different length channel to adjust behavior
+- `cache`
+  - Any implementation of the [SessionCache Interface](https://github.com/nerdswords/yet-another-cloudwatch-exporter/blob/b051535a7b26a8d202f21df5ec14e307cabafe18/pkg/sessions.go#L34-L44)
+  - `exporter.NewSessionCache(config, <fips value>)` would be the default
+  - `<fips value>` is defined by the `fips` CLI flag
+- `observedMetricLabels`
+  - Prometheus requires that all metrics exported with the same key have the same labels
+  - This map will track all labels observed and ensure they are exported on all metrics with the same key in the provided `registry`
+  - You should provide the same instance of this map if you intend to re-use the `registry` between calls
+The update definition also includes an exported array of `Metrics` which includes AWS API call metrics. These can be registered with the provided `registry` if you want them
```
Comment (matthewnolf) [inline README.md]:
```suggestion
The update definition also includes an exported slice of `Metrics` which includes AWS API call metrics. These can be registered with the provided `registry` if you want them
```

**Review by matthewnolf — APPROVED** [review]
Left some _tiny_ nits, looks good

File: README.md (hunk 2 above)
Comment (kgeckhart) [reply README.md]: > A link to GoDoc here may also be nice

Good call, I should probably add a GoDoc as well

**Review by ferruvich — APPROVED** [review]

**Comment by kgeckhart** [comment]
Thanks for the feedback, upstream PR is open https://github.com/nerdswords/yet-another-cloudwatch-exporter/pull/560, I'll close this and will sync it with upstream when it comes through.

**Pull request closed without merging**